### PR TITLE
Two-graph requirement header progress colors and warnings

### PR DESCRIPTION
**Pull request by willespencer**
### Summary 

This pull request continues the two-graph front-end by updating the progress counter/bar/warning for college/major/minor headers. Adapted from @SamChou19815 's #599 and designers [here](https://www.figma.com/file/vz9T397rHdewbB2McmUDEr/Yuxuan-Working-File?node-id=2035%3A1560).

- [x] Set progress counter to represent all fulfilled requirements, even those that are dangerously fulfilled
- [x] Set progress bar color to be blue for safely fulfilled reqs, yellow for dangerously fulfilled
- [x] Add a `ProgressBarWarning` next to the bar to show number of conflicts when they exist. 
- [x] **NEW** Add GK so that yellow progress and new warnings only show when conflicts enabled 

Remaining TODOs:

- [x] Fix warning z-index issue, see Notes.
- [ ] Update CourseCaution warning icon color after #621 merged depending, see Notes.
- [ ] Add the functionality that actually creates conflicted requirements in a future PR.

Depends on #621 for full requirement bar two-graph implementation.

### Test Plan 

To test the below, I updated the function getAllCoursesThatCanPotentiallySatisfyRequirement in requirement-graph-builder-from-user-data.ts locally by adding an edge between a course and a new req in `dangerousRequirementFulfillmentGraph`, and removing the existing edge between that course and the edge it is assigned to in `safeRequirementFulfillmentGraph.` This simulates as if a course has two dangerous requirements it is being double counted for (and thus no safe requirement). To do so, the courseID and requirement name must be hardcoded in (accessed from Firestore and the Requirements debugger respectively).

1. Add a second fulfillment to the CS 4121 course in plan of a CS Major.
image

2. Confirm that then that the progress bar will be yellow, representing the practicum requirement has a conflict. Further confirm the fulfillment number / 8 still counts the practicum requirement being complete despite the conflict. Finally, confirm the warning icon with tooltip shows up next to the fulfillment count.
image

3. Confirm the above also works for compounded requirements (i.e. liberal studies)
image
image

4. **NEW** Confirm that turning on/off GK turns off/on yellow progress bar + warning
image
image

### Notes 

Two design/implementation questions
1. As seen in the screenshots above, the warning tooltip is partially covered by the `SemesterView`, especially on certain breakpoints. Changing the z-index does not seem to easily fix the issue. Any suggestions on how to fix this in the code, or if not, could the warning be placed to the left of `0/8 Total Requirements Inputted on Schedule` instead of to the right?
2. The warning icons for these new conflicts on Figma are a different color than our existing warnings (the new ones match the progress bar color & highlight color we use for conflicts). If this is intentional, should this color also be used on course cards that have conflicts? If so, what if a course has both a conflict and an existing type of warning (like duplicate)?

Once these issues are resolved, this PR can be safely merged without affecting users until conflicts can actually be created.

### CODE REVIEW

**Comment by einc** [comment]
design questions:
1. yes, left is good!
2. yes, it was intentional and can be used for all course cards that have conflicts (and existing types of warnings) for consistency

**Comment by willespencer** [comment]
> 1. yes, left is good!

Done! Thank you :smile

image

> 2. yes, it was intentional and can be used for all course cards that have conflicts (and existing types of warnings) for consistency

That makes sense to me! Can't fully implement this until my other PR #621 is merged, but should be easy to fix right after with the infra I set up (just setting the new `showConflictCaution` flag to true)

**Review by benjamin-shen — APPROVED** [review]
lgtm 🥳

File: src/components/Requirements/RequirementHeader.vue [hunk 1]
```diff
@@ -213,13 +227,14 @@ export default defineComponent({
     },
   },
   computed: {
-    // number of fully fulfilled requirements, note pure self-checks are never fulfilled
-    requirementFulfilled(): number {
+    // number of fully fulfilled requirements, including those with courses that have conflicts
+    // note pure self-checks are never fulfilled
```
[the remark this answers is not in the document]
Comment (willespencer) [reply src/components/Requirements/RequirementHeader.vue:231]: It seems like we do have (mostly) pure self-checks still - there are a few reqs where we do not check at all if they are fulfilled, but courses can still be assigned to them.

image

**Review by benjamin-shen — APPROVED** [review]

**Pull request merged**